Core.ApplicationServices/Organizations/OrganizationRightsService.cs

```diff
@@ -75,6 +87,122 @@ public class OrganizationRightsService : IOrganizationRightsService
             return RemoveRight(right);
         }
 
+        public Maybe<OperationError> RemoveUnitRightsByIds(Guid organizationUuid, Guid unitUuid, IEnumerable<int> rightIds)
+        {
+            using var transaction = _transactionManager.Begin();
+
+            var unitResult = GetOrganizationUnitByUuid(organizationUuid, unitUuid);
+            if (unitResult.Failed)
+            {
+                return unitResult.Error;
+            }
+            
+            var rightsToDelete = new List<OrganizationUnitRight>();
+            foreach (var rightId in rightIds)
+            {
+                var unitRightResult = unitResult.Value.GetRight(rightId);
```

Comment:
Why not call RemoveRight and then make that a `Result<OrganizationUnitRight,OperationError>`
That allows the organizationunit to: 
- check if the right was there
- Remove it if it does
- If not, error
you then collect the result (if successful) and add to your "removal" list